### ISSUE TITLE
Merge PostgreSQL metadata, geometry and Google App Engine metadata exports

### ISSUE THREAD

**Issue by gaurav** [MapofLife/MOL#1]
We're going to clean up our bulkloading process and set it up as:
1. For each shapefile:
   1. Create metadata CSV (for ALL rows in this shapefile)
   2. Bulkload to PostgreSQL (by doing a COPY FROM CSV ...). This will give the rows unique 'id's.
   3. Bulkload to Google App Engine  (by doing a single COPY TO CSV .. WHERE prov=p, coll=c, filename=f)
      1. Run shp2pgis.sql on shapefile -> temp table
   4. Update layers table using nested select magic (INSERT INTO layers WHERE .... FROM (SELECT geom FROM TempTable ...))

This issue is just to document the process we're aiming for and for naming the branch :).

**Comment by gaurav** [comment]
As of @ce6c4ab0615ded0, steps 1-3 are done, pretty much exactly as anticipated. Now we just need to get the shapefile SQL in (steps 4-5), and we're done with this!

**Comment by gaurav** [comment]
Cody and I discovered today that the blank-DBF hack essentially deletes all the information in the SHP file, possibly as it tries to reconcile DBF records with the SHP file.

I tried to fix that in @ad85caee6fdc by replacing pyshp with ogr2ogr - it's a big change, but (1) makes us reliant on only a single program, and (2) ogr2ogr's output has the PostGIS hash in the _first_ column, so we can extract it without fiddling with any of the other fields. This works really well, except for two problems:
1. On my Mac, ogr2ogr somehow keeps the .shp.csv file open even after it's apparently exit()ed. After processing something like 380 shapefiles, it hits a limit of open files and cannot process any more files.
2. It is currently trying to upload the geometries to Google App Engine, which is unnecessary, and fails where the geometry is really long (I'm getting errors at ~131,000 chars).

I would actually like to eventually move to a single ogr2ogr extraction for everything: I remain nervous that if ogr2ogr's CSV and PostGIS outputs get misordered, we would end up matching geometries to the wrong metadata layer, which would be both catastrophic and hard to identify.
